components/ca-vote.js

```diff
@@ -0,0 +1,156 @@
+         * @returns {string} unique string ID for voted-for feature
+         */
+        get id() {
+            return this.getAttribute('id') || '';
+        }
+
+        /**
+         * Set ID under which this vote will be counted.
+         * @param {string} id - unique string ID for voted-for feature
+         */
+        set id(id) {
+            this.setAttribute('id', id || '');
+        }
+
+        /**
+         * Gets key used for recording this vote in local storage to prevent double-voting.
+         * This is defaulted to <code>ca-vote_<i>id</i></code>.
+         * @returns {string} the key used as a double-vote check
+         */
+        get storageKey() {
+            return this._storageKey;
+        }
+
+        /**
+         * Gets the current displayed text in the control.
+         * This property is reflected onto the <code>data-label</code> attribute.
+         * Normally this is managed by the {@link Vote.render} method.
+         * @returns {string} the current text being displayed
+         */
+        get label() {
+            return this.getAttribute('data-label');
+        }
+
+        /**
+         * Sets the current displayed text in the control.
+         * This property is reflected onto the <code>data-label</code> attribute.
+         * @param {string} label - the text to display in the control
+         */
+        set label(label) {
+            this.setAttribute('data-label', label);
+        }
+
+        /**
+         * Gets the current status of the vote button, either <code>disabled</code> or <code>enabled</code>.
+         * This property is reflected onto the <code>data-status</code> attribute and used to disable the click behaviour
+         * and affect the styling.
+         * Normally this is managed by the {@link Vote.render} method.
+         * @returns {string} the current status
+         */
+        get status() {
+            return this.getAttribute('data-status');
+        }
+
+        /**
+         * Sets the current status of the vote button, either <code>disabled</code> or <code>enabled</code>.
+         * This property is reflected onto the <code>data-status</code> attribute and used to disable the click behaviour
+         * and affect the styling.
+         * Normally this is managed by the {@link Vote.render} method.
+         * @param {string} value - the status to apply
+         */
+        set status(value) {
+            this.setAttribute('data-status', value);
+        }
+
+        /**
+         * Renders the ca-vote element
+         * @returns {void} void
+         */
+        render() {
+            clearElement(this);
+
+            if (localStorage[this.storageKey]) {
+                this.status = 'disabled';
+                this.label = 'Thanks for your vote';
+            } else {
+                this.status = 'enabled';
+                this.label = 'Vote for this feature';
+            }
+
+            this.button = createElement(this, 'button', null, this.label);
```

Comment:
I think this obfuscation is unnecessary, but happy to have that discussion come refactoring time.